src/custom-components/ReorderableList.tsx

```diff
@@ -0,0 +1,129 @@
+import { Fragment, JSXElementConstructor, ReactElement, useState } from "react";
+import { Field, FieldProps, Focusable, GamepadButton } from "../deck-components";
+
+export type ReorderableEntry<T> = {
+  label: string,
+  data?:T,
+  position:number
+}
+
+type ListProps<T> = {
+  entries: ReorderableEntry<T>[],
+  onSave: (entries: ReorderableEntry<T>[]) => void,
+  interactables?: JSXElementConstructor<{entry:ReorderableEntry<T>}>,
+  fieldProps?: FieldProps
+}
+
+/**
+ * A component for creating reorderable lists.
+ * 
+ * Implementation example can be found {@link https://github.com/Tormak9970/Component-Testing-Plugin/blob/main/src/testing-window/ReorderableListTest.tsx here}.
+ */
+export function ReorderableList<T>(props: ListProps<T>) {
+  const [entryList, setEntryList] = useState<ReorderableEntry<T>[]>(props.entries.sort((a:ReorderableEntry<T>, b:ReorderableEntry<T>) => a.position - b.position));
+  const [reorderEnabled, setReorderEnabled] = useState<boolean>(false);
+
+  function toggleReorderEnabled(): void {
+    let newReorderValue = !reorderEnabled;
+    setReorderEnabled(newReorderValue);
+
+    if (!newReorderValue){
+      props.onSave(entryList);
+    }
+  }
+
+  return (
+    <Fragment>
+      <style>{`
+        .reorderable-list {
+          width: inherit;
+          height: inherit;
+
+          flex: 1 1 1px;
+          scroll-padding: 48px 0px;
+          display: flex;
+          flex-direction: column;
+          justify-content: flex-start;
+          align-content: stretch;
+        }
+
+        .reorderable-list .custom-buttons {
+          width: inherit;
+          height: inherit;
+          display: inherit;
+        }
+      `}</style>
+      <div className="reorderable-list">
+        <Focusable
+          onSecondaryButton={toggleReorderEnabled}
+          onSecondaryActionDescription={reorderEnabled ? "Save Order" : "Reorder"}
+          onClick={toggleReorderEnabled}>
+          {
+            entryList.map((entry: ReorderableEntry<T>) => (
+              <ReorderableItem listData={entryList} entryData={entry} reorderEntryFunc={setEntryList} reorderEnabled={reorderEnabled} fieldProps={props.fieldProps}>
+                {props.interactables ? <props.interactables entry={entry} /> : null}
+              </ReorderableItem>
+            ))
+          }
+        </Focusable>
+      </div>
+    </Fragment>
+  );
+}
+
+type ListEntryProps<T> = {
+  fieldProps?: FieldProps,
+  listData: ReorderableEntry<T>[],
+  entryData: ReorderableEntry<T>,
+  reorderEntryFunc: CallableFunction,
+  reorderEnabled: boolean,
+  children:ReactElement|null
```

Comment:
You may want to have a Decky maintainer run this through a linter so it matches our Prettier config.

Reply:
ran prettier on newest update